plenigo_sdk/plenigo/builders/CheckoutSnippetBuilder.php

```diff
@@ -67,16 +70,47 @@ public function build($settings = array(), $loginToken = null, $showRegisterFirs
         $requestQueryString = $this->buildCheckoutRequestQueryString($settings);
         $encodedData = $this->buildEncodedData($requestQueryString);
         PlenigoManager::get()->notice($clazz, "Checkout QUERYSTRING:" . $requestQueryString);
-        if (is_null($loginToken)) {
-            if($showRegisterFirst){
-                return sprintf("plenigo.checkout('%s', true);", $encodedData);
-            }else{
-                return sprintf("plenigo.checkout('%s');", $encodedData);
-            }
-        } else {
+
+        $strFunction = "plenigo.checkout";
+        $strFirstParam = "";
+        if ($showRegisterFirst) {
+            $strFirstParam = ", true";
+        }
+        if (!is_null($loginToken)) {
             PlenigoManager::get()->notice($clazz, "Login TOKEN:" . $loginToken);
-            return sprintf("plenigo.checkoutWithRemoteLogin('%s','%s');", $encodedData,$loginToken);
+            $strFunction = "plenigo.checkoutWithRemoteLogin";
+            $strFirstParam = ", '" . $loginToken . "'";
+        }
+        $strSourceURL = "";
+        $strTargetURL = "";
+        if (!is_null($sourceUrl)) {
+            PlenigoManager::get()->notice($clazz, "Source URL:" . $sourceUrl);
+            $strSourceURL = ", '" . $sourceUrl . "'";
+            if (!$showRegisterFirst && is_null($loginToken)) {
+                $strFirstParam = ", false";
+            }
```

Comment:
Is this missing here?
if (is_null($targetUrl)) {
                $targetUrl = ", null";
}